### PR TITLE
Add OSX keystroke to clear RPCConsole

### PR DESCRIPTION
**Pull request by spencerlievens**
Currently only Ctrl-L is mentioned in help, but, CMD-L works on OSX and isn't mentioned.

### CODE REVIEW

**Comment by jonasschnelli** [comment]
ACK 6315a6fd0689c7469f9c266bf5b4f0e337ef79d4

File: src/qt/rpcconsole.cpp [hunk 1]
```diff
@@ -731,7 +731,8 @@ void RPCConsole::clear(bool clearHistory)
         );
 
     message(CMD_REPLY, (tr("Welcome to the %1 RPC console.").arg(tr(PACKAGE_NAME)) + "<br>" +
-                        tr("Use up and down arrows to navigate history, and <b>Ctrl-L</b> to clear screen.") + "<br>" +
+                        tr("Use up and down arrows to navigate history") + "<br>" +
+                        tr("To clear screen use <b>Ctrl-L</b> on Windows/Linux or <b>⌘CMD-L</b> on OSX") + "<br>" +
```
Comment (laanwj) [inline src/qt/rpcconsole.cpp]: I'm not usually a person to say this but: let's make this platform dependent instead of enumerating the platforms here.

Comment (spencerlievens) [inline src/qt/rpcconsole.cpp]: ifdef it?

**Comment by spencerlievens** [comment]
Improved the code to use an ifdef for OSX instead.

File: src/qt/rpcconsole.cpp [hunk 2]
```diff
@@ -731,8 +731,12 @@ void RPCConsole::clear(bool clearHistory)
...
 
     message(CMD_REPLY, (tr("Welcome to the %1 RPC console.").arg(tr(PACKAGE_NAME)) + "<br>" +
-                        tr("Use up and down arrows to navigate history, and <b>Ctrl-L</b> to clear screen.") + "<br>" +
-                        tr("Type <b>help</b> for an overview of available commands.")) +
+#ifdef Q_OS_MAC
+                        tr("Use up and down arrows to navigate history, and <b>⌘cmd-L</b> to clear screen.") + "<br>" +
+#else
+                        tr("Use up and down arrows to navigate history, and <b>Ctrl-L</b> to clear screen.") + "<br>" +		
+#endif
+			tr("Type <b>help</b> for an overview of available commands.")) +
```
Comment (laanwj) [inline src/qt/rpcconsole.cpp]: Possible suggestion:
```c++
#ifdef Q_OS_MAC
QString clearKeyStr = "<b>⌘cmd-L</b>";
#else
QString clearKeyStr = "<b>Ctrl-L</b>";
#endif
...
tr("Use up and down arrows to navigate history, and %1 to clear screen.").arg(clearKeyStr) + "<br>" +
```
After all:
- The key is not to be translated
- Also gets rid of the HTML formatting in the translation string in one go

Comment (jonasschnelli) [inline src/qt/rpcconsole.cpp]: If we are already bikeshedding:
Alternative suggestion would be adding `QString generalFunctionKey()` to platformstyles.cpp.
Then use something like `tr("Use up and down arrows to navigate history, and %1-L to clear screen.").arg(platformStyle->generalFunctionKey())`

Comment (spencerlievens) [inline src/qt/rpcconsole.cpp]: That is much tidier than my attempt 👍 

#ifdef Q_OS_MAC
QString clearKeyStr = "<b>⌘cmd-L</b>";
#else
QString clearKeyStr = "<b>Ctrl-L</b>";
#endif

Comment (laanwj) [inline src/qt/rpcconsole.cpp]: Yes, adding it to platformStyle would be another step further, I didn't want to suggest that here to not make life too difficult for @spencerlievens while also not making life harder for translators by adding another message that is almost-the-same.

Comment (spencerlievens) [inline src/qt/rpcconsole.cpp]: Yeah adding to platform style is certainly the more "correct" path, but induces a lot of work.

**Comment by spencerlievens** [comment]
@laanwj When I find some time I will add to platformstyle.cpp and update the translation files as it isn't difficult, just time consuming.

File: src/qt/rpcconsole.cpp [hunk 3]
```diff
@@ -730,9 +730,15 @@ void RPCConsole::clear(bool clearHistory)
             ).arg(fixedFontInfo.family(), QString("%1pt").arg(consoleFontSize))
         );
 
+#ifdef Q_OS_MAC
+    QString clsKey = "<b>⌘cmd-L</b>";
+#else
+    QString clsKey = "<b>Ctrl-L</b>";
+#endif
+
```
Comment (paveljanik) [inline src/qt/rpcconsole.cpp]: There is a TAB on this otherwise empty line.

File: src/qt/rpcconsole.cpp [hunk 4]
```diff
@@ -730,9 +730,15 @@ void RPCConsole::clear(bool clearHistory)
             ).arg(fixedFontInfo.family(), QString("%1pt").arg(consoleFontSize))
         );
 
+#ifdef Q_OS_MAC
+    QString clsKey = "<b>⌘cmd-L</b>";
```
Comment (paveljanik) [inline src/qt/rpcconsole.cpp]: Can you please move this HTML down to `arg(clsKey)`? E.g.
```C++
arg("<b>"+clsKey+</b>)
```
or something and remove it from here?

File: src/qt/rpcconsole.cpp [hunk 5]
```diff
@@ -730,9 +730,15 @@ void RPCConsole::clear(bool clearHistory)
...
+#else
+    QString clsKey = "<b>Ctrl-L</b>";
+#endif
+	
     message(CMD_REPLY, (tr("Welcome to the %1 RPC console.").arg(tr(PACKAGE_NAME)) + "<br>" +
-                        tr("Use up and down arrows to navigate history, and <b>Ctrl-L</b> to clear screen.") + "<br>" +
-                        tr("Type <b>help</b> for an overview of available commands.")) +
+                        tr("Use up and down arrows to navigate history, and %1 to clear screen.").arg(clsKey) + "<br>" +
+			tr("Type <b>help</b> for an overview of available commands.")) +
```
Comment (paveljanik) [inline src/qt/rpcconsole.cpp]: Spaces replaced by TABS? Why?

**Review by paveljanik — CHANGES_REQUESTED** [review]
And please squash.

File: src/qt/rpcconsole.cpp (hunk 5 above)
Comment (spencerlievens) [inline src/qt/rpcconsole.cpp]: Urgh that was my editor.

File: src/qt/rpcconsole.cpp (hunk 3 above)
Comment (spencerlievens) [inline src/qt/rpcconsole.cpp]: Done

File: src/qt/rpcconsole.cpp (hunk 4 above)
Comment (spencerlievens) [inline src/qt/rpcconsole.cpp]: Done

File: src/qt/rpcconsole.cpp [hunk 6]
```diff
@@ -730,8 +730,14 @@ void RPCConsole::clear(bool clearHistory)
...
 
+#ifdef Q_OS_MAC
+    QString clsKey = "⌘cmd-L";
+#else
+    QString clsKey = "Ctrl-L";
+#endif
+
     message(CMD_REPLY, (tr("Welcome to the %1 RPC console.").arg(tr(PACKAGE_NAME)) + "<br>" +
-                        tr("Use up and down arrows to navigate history, and <b>Ctrl-L</b> to clear screen.") + "<br>" +
+                        tr("Use up and down arrows to navigate history, and %1 to clear screen.").arg("<b>"+clsKey+"<b>") + "<br>" +
```
Comment (paveljanik) [inline src/qt/rpcconsole.cpp]: `</b>`

**Comment by spencerlievens** [comment]
@paveljanik thats what happens when you edit via mobile and online on GitHub

**Comment by spencerlievens** [comment]
@laanwj @jonasschnelli @paveljanik 

All squashed and good to go ;)

**Comment by paveljanik** [comment]
In the help of e.g. Terminal and Safari, I see Command (⌘)-L notation.

**Comment by spencerlievens** [comment]
@paveljanik This is true, but I wanted to spell it out for the user and make it clear as day.

**Comment by spencerlievens** [comment]
I realised that this work was not done on a Bitcoin fork and the fork of Dash that the work had been done on I have subsequently deleted, the new PR for this is #10362

**Pull request closed without merging**